src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,272 @@
+
+    if (this.verbose) {
+      logger.logToStderr(`Starting Batch : ${batchNumber}`);
+    }
+    Promise
+      .all(this.batchedPromises[batchNumber])
+      .then((data: any) => {
+        this.completeAuditReports.push(data);
+        if (batchNumber < this.batchedPromises.length - 1) {
+          this.getBatchedAuditlogData(logger, ++batchNumber, resolve, reject)
+        }
+        else {
+          resolve();
+        }
+      }, err => { return Promise.reject(err) });
+  }
+
+  private getAuditLogReportforSingleContentURL(auditURL: string): Promise<AuditlogReport[]> {
+    const requestOptions: any = {
+      url: auditURL,
+      headers: {
+        accept: 'application/json;'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<AuditlogReport[]>(requestOptions);
+  }
+
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-c, --contentType <contentType>',
+        description: 'Audit content type of logs to be retrieved, should be one of the following: AzureActiveDirectory, Exchange, SharePoint, General, DLP'
+      },
+      {
+        option: '-s, --startTime [startTime]',
+        description: 'Start time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      },
+      {
+        option: '-e, --endTime [endTime]',
+        description: 'End time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if ((<any>AuditContentTypes)[args.options.contentType] === undefined) {
+      return `${args.options.contentType} is not a valid value for the contentType option. Allowed values are ${Object.keys(AuditContentTypes).join(' | ')}`;
+    }
+
+    if (args.options.startTime || args.options.endTime) {
+      if (!(args.options.startTime)) {
+        return `Please specify a starttime`;
+      }
+
+      const parsedStartTime = Date.parse(args.options.startTime as string)
+      if (args.options.startTime && !(!parsedStartTime) !== true) {
+        return `${args.options.startTime} is not a valid startTime. Provide the date in one of the following formats:
+      'YYYY-MM-DD'
+      'YYYY-MM-DDThh:mm'
+      'YYYY-MM-DDThh:mmZ'
+      'YYYY-MM-DDThh:mm±hh:mm'`;
+      }
+
+      const StartdateTodayDifference: number = (new Date().getTime() - parsedStartTime) / (1000 * 60 * 60 * 24);
+      if (StartdateTodayDifference > 7) {
+        return `Start time should be no more than 7 days in the past`;
+      }
+
+      if (args.options.endTime) {
+        const parsedEndTime: number = Date.parse(args.options.endTime as string);
+        if (!(!parsedEndTime) !== true) {
```

Comment:
What is `!(!parsedEndTime) !== true` meant to check? Can we clarify it?

Reply:
I was using the same validation approach which is used before in the codebase which is referred [HERE](https://github.com/pnp/cli-microsoft365/blob/bc41cb1f966245111751544273f3eafbf8db1afa/src/m365/spo/commands/list/list-webhook-set.ts#L146). 

Just thought, I would re-use the similar validation technique which is used before rather than re-inventing the wheel. 

Now instead of that, I have given below validation which I guess would be the optimal one,

```
        if (isNaN(parsedEndTime)) {
          return `${args.options.endTime} is not a valid endTime. Provide the date in one of the following formats:
        'YYYY-MM-DD'
        'YYYY-MM-DDThh:mm'
        'YYYY-MM-DDThh:mmZ'
        'YYYY-MM-DDThh:mm±hh:mm'`;
        }
```

Kindly let me know if the approach is not optimal.